File: CSHARP/Cap07/c07ex01.cs
// c07ex01.cs

using System;
using static System.Console;

class Program
{
    static void RotinadeAdicao()
    {
        float a1, b1, r1;
        WriteLine();
        WriteLine("Rotina de Adicao");
        WriteLine();
        Write("Entre o 1o. valor: ");
        a1 = float.Parse(ReadLine());
        Write("Entre o 2o. valor: ");
        b1 = float.Parse(ReadLine());
        WriteLine();
        r1 = a1 + b1;
        WriteLine($"O resultado da operacao equivale a: {r1}");
        WriteLine();
    }

    static void RotinadeSubtracao()
    {
        float a2, b2, r2;
        WriteLine();
        WriteLine("Rotina de Subtracao");
        WriteLine();
        Write("Entre o 1o. valor: ");
        a2 = float.Parse(ReadLine());
        Write("Entre o 2o. valor: ");
        b2 = float.Parse(ReadLine());
        WriteLine();
        r2 = a2 - b2;
        WriteLine($"O resultado da operacao equivale a: {r2}");
        WriteLine();
    }

    static void RotinaDeMultiplicacao()
    {
        float a3, b3, r3;
        WriteLine();
        WriteLine("Rotina de Multiplicacao");
        WriteLine();
        Write("Entre o 1o. valor: ");
        a3 = float.Parse(ReadLine());
        Write("Entre o 2o. valor: ");
        b3 = float.Parse(ReadLine());
        WriteLine();
        r3 = a3 * b3;
        WriteLine($"O resultado da operacao equivale a: {r3}");
        WriteLine();
    }

    static void RotinaDeDivisao()
    {
        float a4, b4, r4;
        WriteLine();
        WriteLine("Rotina de Divisao");
        WriteLine();
        Write("Entre o 1o. valor: ");
        a4 = float.Parse(ReadLine());
        Write("Entre o 2o. valor: ");
        b4 = float.Parse(ReadLine());
        WriteLine();
        if (b4 == 0)
            WriteLine("O resultado da operacao equivale a: ERRO");
        else
        {
            r4 = a4 / b4;
            WriteLine($"O resultado da operacao equivale a: {r4}");
        }
        WriteLine();
    }

    static void Main(string[] args)
    {
        int opcao = 0;
        while (opcao != 5)
        {
            WriteLine("CALCULADORA - V1");
            WriteLine();
            WriteLine("[1] - Adicao");
            WriteLine("[2] - Subtracao");
            WriteLine("[3] - Multiplicacao");
            WriteLine("[4] - Divisao");
            WriteLine("[5] - Fim de Programa");
            WriteLine();
            Write("Escolha uma opcao: ");
            opcao = int.Parse(ReadLine());

            if (opcao != 5)
            {
                switch (opcao)
                {
                    case 1:
                        RotinadeAdicao();
                        break;
                    case 2:
                        RotinadeSubtracao();
                        break;
                    case 3:
                        RotinaDeMultiplicacao();
                        break;
                    case 4:
                        RotinaDeDivisao();
                        break;
                    default:
                        WriteLine();
                        WriteLine("Opcao invalida - Tente novamente");
                        WriteLine();
                        break;
                }
            }
        }
    }
}

File: SCALA/Cap07/c07ex01.scala
// c07ex01.scala

object Main {
  def rotAdicao(): Unit = {
    println()
    println("Rotina de Adicao")
    println()
    print("Entre o 1o. valor: ")
    var a1 = readFloat()
    print("Entre o 2o. valor: ")
    var b1 = readFloat()
    println()
    var r1 = a1 + b1
    println(s"O resultado da operacao equivare a: $r1")
    println()
  }

  def rotSubtracao(): Unit = {
    println()
    println("Rotina de Subtracao")
    println()
    print("Entre o 1o. valor: ")
    var a2 = readFloat()
    print("Entre o 2o. valor: ")
    var b2 = readFloat()
    println()
    var r2 = a2 - b2
    println(s"O resultado da operacao equivare a: $r2")
    println()
  }

  def rotMultiplicacao(): Unit = {
    println()
    println("Rotina de Multiplicacao")
    println()
    print("Entre o 1o. valor: ")
    var a3 = readFloat()
    print("Entre o 2o. valor: ")
    var b3 = readFloat()
    println()
    var r3 = a3 * b3
    println(s"O resultado da operacao equivare a: $r3")
    println()
  }

  def rotDivisao(): Unit = {
    println()
    println("Rotina de Divisao")
    println()
    print("Entre o 1o. valor: ")
    var a4 = readFloat()
    print("Entre o 2o. valor: ")
    var b4 = readFloat()
    println()
    if (b4 == 0)
      println("O resultado da operacao equivare a: ERRO")
    else {
      var r4 = a4 / b4
      println(s"O resultado da operacao equivare a: $r4")
    }
    println()
  }

  def main(args: Array[String]): Unit = {
    var opcao = 0
    while (opcao != 5) {
      println("CALCULADORA - V1")
      println()
      println("[1] - Adicao")
      println("[2] - Subtracao")
      println("[3] - Multiplicacao")
      println("[4] - Divisao")
      println("[5] - Fim de Programa")
      println()
      print("Escolha uma opcao: ")
      opcao = readInt()

      if (opcao != 5) {
        opcao match {
          case 1 => rotAdicao()
          case 2 => rotSubtracao()
          case 3 => rotMultiplicacao()
          case 4 => rotDivisao()
          case _ =>
            println()
            println("Opcao invarida - Tente novamente")
            println()
        }
      }
    }
  }
}
